pkg/controller/container-runtime-config/helpers.go

```diff
@@ -244,22 +241,176 @@ func updateCRIOConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfiguratio
 	return newData.Bytes(), nil
 }
 
-func updateRegistriesConfig(data []byte, internalInsecure, internalBlocked []string) ([]byte, error) {
-	tomlConf := new(tomlConfigRegistries)
-	if _, err := toml.Decode(string(data), tomlConf); err != nil {
+// prefixMatchesRegistry returns true if a prefix value (as in sysregistriesv2.Registry.Prefix / sysregistriesv2.Endpoint.Location)
+// matches a host[:port] value in reg.
+func prefixMatchesRegistry(prefix, reg string) bool {
+	// NOTE: "prefix" is the sysregistries.Registry name of the object, and it can be a host[:port] or a namespace within;
+	// i.e. we want to return true if "reg" is a prefix of "prefix"!
+	if reg == prefix {
+		return true
+	}
+	if len(prefix) > len(reg) {
+		return strings.HasPrefix(prefix, reg) && prefix[len(reg)] == '/'
+	}
+	return false
+}
+
+// disjointOrderedMirrorSets processes icspRules and returns the resulting disjoint sets of mutual mirrors, each ordered in consistently with
+// the icspRules preference order (if possible)
+// This exists to merge different mirror sets, e.g. given mirror sets (B, C) and (A, B), it will combine them into a single (A, B, C) set.
+func disjointOrderedMirrorSets(icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy) ([][]string, error) {
+	// Collect all mirror sets into a flat array to abstract the rest of the code from the ImageContentSourcePolicy format
+	mirrorSets := []*apioperatorsv1alpha1.RepositoryDigestMirrors{}
+	for _, icsp := range icspRules {
+		for i := range icsp.Spec.RepositoryDigestMirrors {
+			set := &icsp.Spec.RepositoryDigestMirrors[i]
+			if len(set.Sources) < 2 {
```

Comment:
why not if len(set.Sources) >= 2 { mirrorSets....}

Reply:
This way the comment about the `len < 2` case can be placed in the matching code path; using `len >= 2` would force the comment to be above the `if`, and redundantly describe the condition.

I don’t care at all strongly about this.

Reply:
i think it would be better to open a github issue to cover this FIXME since they tend to get lost in the shuffle.

Reply:
Yeah, filed https://github.com/openshift/machine-config-operator/issues/949 .